llmfoundry/callbacks/fdiff_callback.py

```diff
@@ -47,10 +47,12 @@ def batch_end(self, state: State, logger: Logger):
     def eval_end(self, state: State, logger: Logger):
         if self.diff_eval_metrics:
             evaluator = state.dataloader_label
-            metrics = list(state.eval_metrics[evaluator].keys())  # type: ignore
+            assert evaluator is not None, 'dataloader should have been set'
```

Comment:
raise error instead of assert

Reply:
these asserts are ok because they are just for pyright, logically the `state.dataloader_label` should never be `None` at this point.